File: leetcode_75/dp_1d/n-th_tribonacci_number_1137.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
    public int tribonacci(int n) {
        List<Integer> ll = new ArrayList<>(Collections.nCopies(n + 3, 0));
        ll.set(1, 1);
        ll.set(2, 1);
        for (int i = 3; i <= n; i++) {
            int sum = ll.get(i - 1) + ll.get(i - 2) + ll.get(i - 3);
            ll.set(i, sum);
        }
        return ll.get(n);
    }
}

File: leetcode_75/dp_1d/n-th_tribonacci_number_1137.py
class Solution:
    def tribonacci(self, n: int) -> int:
        ll = [0]*(n+3)
        ll[0],ll[1],ll[2] = 0, 1, 1
        for n in range(3, n+1):
            ll[n] = ll[n-1] + ll[n-2] + ll[n-3]
        return ll[n]